packages/coverage-v8/src/provider.ts

```diff
@@ -497,27 +486,41 @@ export class V8CoverageProvider extends BaseCoverageProvider implements Coverage
 
   private async convertCoverage(
     coverage: RawCoverage,
-    projectName?: ProjectName,
-    transformMode?: 'web' | 'ssr',
+    project: WorkspaceProject = this.ctx.getCoreWorkspaceProject(),
+    transformMode?: keyof CoverageFilesByTransformMode,
   ): Promise<CoverageMap> {
-    const viteNode
-      = this.ctx.projects.find(project => project.getName() === projectName)
-        ?.vitenode || this.ctx.vitenode
     const fetchCache = transformMode
-      ? viteNode.fetchCaches[transformMode]
-      : viteNode.fetchCache
+      ? project.vitenode.fetchCaches[transformMode === 'browser' ? 'web' : transformMode]
+      : project.vitenode.fetchCache
     const transformResults = normalizeTransformResults(fetchCache)
 
-    const scriptCoverages = coverage.result.filter(result =>
-      this.testExclude.shouldInstrument(fileURLToPath(result.url)),
-    )
+    async function onTransform(filepath: string) {
+      if (transformMode === 'browser' && project.browser) {
+        const result = await project.browser.vite.transformRequest(filepath)
```

Comment:
If filepath is absolute, this will load a duplicate module because IDs are different - if you want to take the value from the cache, the path should be relative to root (`/src/t.ts` instead of `/home/project/src/t.ts`)

Reply:
Good catch! Added change to use root relative paths here.